Crash when screen is locked and monitor goes into power-saving mode

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I found a [similar crash report](https://github.com/zed-industries/zed/issues/5951) but since that was for Mac and this is for Linux, I figured I'd open a new one.

To reproduce, all I have to do is start Zed and lock my screen. If I immediately unlock my machine Zed is still running with no problems, but if I am gone long enough for my external monitor to go into power-saving mode, then when I unlock my machine Zed is gone. This happens the vast majority of the time, though there have been a small handful of times (i.e. 3 or 4 times over the last few months) Zed was still running when I unlocked my machine.

The attached log is from about 10 minutes after I locked my computer, which is my current timeout for power-saving mode.

### Environment

Zed: v0.143.4 (Zed Preview)
OS: Linux X11 ubuntu 24.04
Memory: 31 GiB
Architecture: x86_64

My laptop is a Dell Precision 5570 with a NVIDIA RTX A1000 Laptop GPU/PCIe/SSE2. I am using proprietary drivers (`nvidia-driver-535`). I have it plugged in to an external 4K monitor which is where Zed runs.

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
2024-07-09T12:08:35-06:00 [ERROR] {
  "thread": "main",
  "payload": "GPU hung",
  "location_data": {
    "file": "crates/gpui/src/platform/blade/blade_renderer.rs",
    "line": 415
  },
  "backtrace": [
    "core::panicking::panic_fmt",
    "gpui::platform::blade::blade_renderer::BladeRenderer::draw",
    "<gpui::platform::linux::x11::window::X11Window as gpui::platform::PlatformWindow>::draw",
    "gpui::window::Window::new::{{closure}}::{{closure}}",
    "gpui::window::Window::new::{{closure}}",
    "<core::cell::RefCell<calloop::sources::DispatcherInner<S,F>> as calloop::sources::EventDispatcher<Data>>::process_events",
    "<gpui::platform::linux::x11::client::X11Client as gpui::platform::linux::platform::LinuxClient>::run",
    "gpui::platform::linux::platform::<impl gpui::platform::Platform for P>::run",
    "gpui::app::App::run",
    "zed::main",
    "std::sys_common::backtrace::__rust_begin_short_backtrace",
    "std::rt::lang_start::{{closure}}",
    "std::rt::lang_start_internal",
    "main",
    "__libc_start_call_main",
    "__libc_start_main_impl",
    "_start"
  ],
  "app_version": "0.143.4",
  "release_channel": "Zed Preview",
  "os_name": "Linux X11",
  "os_version": "ubuntu 24.04",
  "architecture": "x86_64",
  "panicked_on": 1720548515021,
  "installation_id": "278045ac-2305-49c3-8f44-e0c07dea5bc2",
  "session_id": "14f75967-78d4-451c-b38a-0859c78c996b"
}
<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/gpui/src/platform/blade/blade_renderer.rs
+++ b/crates/gpui/src/platform/blade/blade_renderer.rs
@@ -18,7 +18,7 @@ use blade_graphics as gpu;
 use blade_util::{BufferBelt, BufferBeltDescriptor};
 use std::{mem, sync::Arc};
 
-const MAX_FRAME_TIME_MS: u32 = 1000;
+const MAX_FRAME_TIME_MS: u32 = 10000;
 
 #[cfg(target_os = "macos")]
 pub type Context = ();
@@ -412,7 +412,8 @@ impl BladeRenderer {
     fn wait_for_gpu(&mut self) {
         if let Some(last_sp) = self.last_sync_point.take() {
             if !self.gpu.wait_for(&last_sp, MAX_FRAME_TIME_MS) {
-                panic!("GPU hung");
+                log::error!("GPU hung");
+                while !self.gpu.wait_for(&last_sp, MAX_FRAME_TIME_MS) {}
             }
         }
     }

